tests/unit/testsuite/Bolt/Boltpay/Controller/Traits/WebHookTraitTest.php

```diff
@@ -0,0 +1,441 @@
+     */
+    private $layout;
+
+    /**
+     * @var PHPUnit_Framework_MockObject_MockObject|Bolt_Boltpay_Helper_Data Mocked instance of Bolt helper
+     */
+    private $helperMock;
+
+    /**
+     * Unregister helper set from previous tests
+     */
+    public static function setUpBeforeClass()
+    {
+        Mage::unregister('_helper/boltpay');
+    }
+
+    /**
+     * Setup test dependencies, called before each test
+     */
+    public function setUp()
+    {
+        Mage::app('default');
+        $this->currentMock = $this->getMockBuilder('Bolt_Boltpay_Controller_Traits_WebHookTrait')
+            ->setMethods(array('getRequest', 'getLayout', 'getResponse'))
+            ->disableOriginalConstructor()
+            ->disableOriginalClone()
+            ->disableArgumentCloning()
+            ->getMockForTrait();
+
+        $this->request = $this->getMockBuilder('Mage_Core_Controller_Request_Http')
+            ->setMethods(array('isAjax', 'getParam'))
+            ->getMock();
+
+        $this->layout = $this->getMockBuilder('Mage_Core_Model_Layout')
+            ->setMethods(array('setDirectOutput'))
+            ->getMock();
+
+        $this->response = $this->getMockBuilder('Mage_Core_Controller_Response_Http')
+            ->setMethods(
+                array(
+                    'setHeader',
+                    'setBody',
+                    'setException',
+                    'clearAllHeaders',
+                    'clearBody',
+                    'setHttpResponseCode',
+                    'sendHeaders',
+                    'sendResponse'
+                )
+            )
+            ->getMock();
+
+        $this->helperMock = $this->getMockBuilder('Bolt_Boltpay_Helper_Data')
+            ->setMethods(
+                array('setResponseContextHeaders', 'verify_hook', 'notifyException')
+            )
+            ->getMock();
+
+        Mage::register('_helper/boltpay', $this->helperMock);
+
+        Mage::getModel('boltpay/observer')->initializeBenchmarkProfiler();
+
+        $this->currentMock->method('getLayout')->willReturn($this->layout);
+        $this->currentMock->method('getRequest')->willReturn($this->request);
+        $this->currentMock->method('getResponse')->willReturn($this->response);
+    }
+
+    /**
+     * Cleanup changes made by tests
+     */
+    protected function tearDown()
+    {
+        Mage::unregister('_helper/boltpay');
+        self::$_fastcgiFinishRequestCalled = false;
+        unset($_SERVER['HTTP_X_BOLT_HMAC_SHA256']);
+        Bolt_Boltpay_StreamHelper::restore();
+    }
+
+    /**
+     * @test
+     * Pre-dispatch method without validating signature
+     *
+     * @covers Bolt_Boltpay_Controller_Traits_WebHookTrait::preDispatch
+     */
+    public function preDispatch_withRequestSignatureValidationDisabled_shouldNotTryToVerifyHook()
+    {
+        $this->response->expects($this->once())->method('clearAllHeaders')->willReturnSelf();
```

Comment:
@Leon in the setUp method you mocked 8 methods of the response, but in the test you set behaviour only on three methods. This is inconsistent

Reply:
Is there a real point to this comment or worth the time to answer?  The other 5 methods will then return null and are not relevant to this test.  So what is your point?